packages/cli/src/zosfiles/ZosFilesBase.handler.ts

```diff
@@ -39,34 +39,35 @@ export abstract class ZosFilesBaseHandler implements ICommandHandler {
      *
      * @returns {Promise<void>}
      */
-    public async process(commandParameters: IHandlerParameters) {
+    public async process(commandParameters: IHandlerParameters): Promise<void> {
         const sessCfg: ISession = ZosmfSession.createSessCfgFromArgs(
             commandParameters.arguments
         );
         const sessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(
-            sessCfg, commandParameters.arguments, {parms: commandParameters}
+            sessCfg, commandParameters.arguments, { parms: commandParameters }
         );
-
         const session = new Session(sessCfgWithCreds);
-        const response = await this.processWithSession(commandParameters, session);
-
-        commandParameters.response.progress.endBar(); // end any progress bars
-        // Print out the response
-        if (response.commandResponse) {
-            commandParameters.response.console.log(response.commandResponse);
-        }
-
-        // Return as an object when using --response-format-json
-        commandParameters.response.data.setObj(response);
-
-        // Ensure error gets thrown if request was unsuccessful.
-        // Sometimes it is useful to delay throwing an error until the end of the handler is
-        // reached, for example the upload API needs to return an API response even when it fails.
-        if (!response.success && response.commandResponse) {
-            throw new ImperativeError({
-                msg: response.errorMessage || response.commandResponse
-            });
+        try {
+            const response = await this.processWithSession(commandParameters, session);
+            if (!response.success && response.commandResponse) {
+                throw new ImperativeError({
+                    msg: response.errorMessage || response.commandResponse
+                });
+            }
+            if (response.commandResponse) {
+                commandParameters.response.console.log(response.commandResponse);
+            }
+            commandParameters.response.data.setObj(response);
+        } catch (error) {
+            if (commandParameters.arguments.ignoreNotFound && error.errorCode === '404') {
+                commandParameters.response.data.setObj({ success: true });
+            } else {
+                throw new ImperativeError({
+                    msg: error.mMessage
```

Comment:
Please add the original error to causeErrors.

Reply:
i guess i didnt really know what to do here because response isnt accessible here because of the catch block so i cant really just like revert back @awharn